lib/mongoose_push/service/apns/state.ex

```diff
@@ -0,0 +1,61 @@
+defmodule MongoosePush.Service.APNS.State do
+  @moduledoc """
+  Module for storing state of apns APNS configurations, namely default topics
+  """
+  use GenServer
+  require Logger
+  alias MongoosePush.Service.APNS.Certificate
+
+  def start_link(arg) do
+    GenServer.start_link(__MODULE__, arg)
+  end
+
+  @impl true
+  def init([pool_configs]) do
+    setup_default_topics(pool_configs)
+    {:ok, %{}}
+  end
+
+  defp setup_default_topics(pool_configs) do
+    default_topics =
+      Enum.map(pool_configs, fn {name, config} ->
+        {name, setup_default_apns_topic(config)[:default_topic]}
+      end)
+
+    :ets.new(:apns_state, [:set, :public, :named_table])
```

Comment:
If we ever used this function more then once, it would fail. I suggest moving `:ets.new` to `init/1`.